app/database/user_item_store.go

```diff
@@ -0,0 +1,243 @@
+package database
+
+import (
+	"database/sql"
+	"fmt"
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+// UserItemStore implements database operations on `users_items`
+type UserItemStore struct {
+	*DataStore
+}
+
+type groupItemPair struct {
+	idGroup int64
+	idItem  int64
+}
+
+// ComputeAllUserItems recomputes fields of users_items
+func (s *UserItemStore) ComputeAllUserItems() (err error) {
+	defer func() {
+		if p := recover(); p != nil {
+			switch e := p.(type) {
+			case runtime.Error, *strconv.NumError:
+				panic(e)
+			default:
+				err = p.(error)
+			}
+		}
+	}()
```

Comment:
We should not handle panics with runtime.Error, because they are caused by errors in the code (such as out-of-bounds array accesses). `*strconv.NumErr` are errors raised by strconv.ParseInt(). We just re-raise them.